### PR TITLE
fix: Scale QK in vision/attention

### PR DESCRIPTION
**Pull request by changlan**
`WindowAttention` layer in vision/attention.py overrides the forward calls directly so we need to scale the QK before computing logits.

### CODE REVIEW

File: axlearn/vision/attention.py [hunk 1]
```diff
@@ -214,6 +214,16 @@ def forward(
...
         self.vlog(3, "atten.v_proj=%s", v_proj.sum())
+
+        # Scale query and key.
+        key_positions = jnp.arange(k_proj.shape[1])[None, :]
+        q_proj, k_proj = self._scale_qk(
+            q_proj=q_proj,
+            k_proj=k_proj,
+            query_positions=query_positions,
+            key_positions=key_positions,
+        )
```
Comment (ruomingp) [inline axlearn/vision/attention.py:225]: Is there a unittest that can be added?

**Review by ruomingp — COMMENTED** [review]
Thanks! Please also see my comments on the internal PR.

File: axlearn/vision/attention.py (hunk 1 above)
Comment (changlan) [reply axlearn/vision/attention.py]: Done - I added a unit test.

**Comment by changlan** [comment]
Gentle ping @ruomingp

**Review by ruomingp — APPROVED** [review]
